documentation/scripts/documentation_compiler.ts

```diff
@@ -128,6 +129,29 @@ function compileSnippetEmbed(codeFile: string) {
   fs.writeFileSync(path.join(snippetDirPath, `${snippetFileName}.html`), exampleSnippetEmbed);
 }
 
+function formatCodeSnippet(code: string, removePlaceholders=false) {
```

Comment:
I don't quite understand what this is doing--is this allowing the snippets to have custom syntax that is then transformed into Monaco syntax? If so why not just use the Monaco syntax directly? I'm probably misunderstanding.

Reply:
The Monaco syntax is `${1:Placeholder}`. That works fine in strings, but if you are trying to use it as a variable name or object key it's invalid typescript. I like the fact that all of our snippets are `.ts` files and compiled with the SDK and didn't want to loose that. So I created a few variants on the format that are also valid identifiers, like `$1$Placeholder$`, etc.

Reply:
Ah I see. Mind adding essentially this explanation to a comment above this function then?

Reply:
Done.